dali/kernels/imgproc/convolution/laplacian_cpu.h

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_CONVOLUTION_LAPLACIAN_CPU_H_
+#define DALI_KERNELS_IMGPROC_CONVOLUTION_LAPLACIAN_CPU_H_
+
+#include <memory>
+
+#include "dali/core/tensor_view.h"
+#include "dali/kernels/common/utils.h"
+#include "dali/kernels/imgproc/convolution/separable_convolution_cpu.h"
+#include "dali/kernels/imgproc/convolution/convolution_cpu.h"
+#include "dali/kernels/kernel.h"
+#include "dali/kernels/scratch.h"
+#include "dali/pipeline/util/operator_impl_utils.h"
+
+namespace dali {
+namespace kernels {
+
+namespace laplacian {
+
+using namespace conv_transform;  // NOLINT
+
+/**
+ * @brief Computes convolution to obtain partial derivative in one of the dimensions.
+ * Convolution consits of `axes` windows, each to convolve along one dimension of the input data,
+ * where `deriv_axis`-th window is supposed to compute partial derivative along that axis,
+ * whereas the remaining windows should perform smoothing. It is assumed that smoothing
+ * window of size 1 must be equal to `[1]`, this way, if window sizes in non-derivative directions
+ * are one, the smoothing convolutions can be skipped and only a single one-dimensional
```

Comment:
```suggestion
 * window of size 1 must be equal to `[1]`. This way, if window sizes in non-derivative directions
 * are one, the smoothing convolutions can be skipped and only a single one-dimensional
```
And I feel there's something wrong with the latter sentence:
> This way, if window sizes in non-derivative directions are one [...]

Should it be `is one`? Or maybe `is [1]`?

Reply:
For 3D case there are in fact two smoothing window sizes for each partial derivative. This optimization handles the case where all the smoothing window sizes are 1, hence "are".